### PR TITLE
Folder class

### PR DESCRIPTION
**Pull request by AnthonySteele**
Make (some of) the file operations injectable
Progress towards full IoC

### CODE REVIEW

File: NuKeeper/GithubEngine.cs [hunk 1]
```diff
@@ -24,30 +26,31 @@ public class GithubEngine
...
             var repositories = await _repositoryDiscovery.GetRepositories();
 
             foreach (var repository in repositories)
             {
                 try
                 {
-                    var repositoryUpdater = new RepositoryUpdater(
-                        _updatesLookup, _github, git, _logger,
-                        tempDir, _updateSelection, repository);
+                    var repositoryUpdater = new RepositoryUpdater(_github, git, _updatesLookup, _updateSelection, tempFolder, _logger, repository);
```
Comment (cohen990) [inline NuKeeper/GithubEngine.cs]: worth adding whitespace formatting just to make it easier to read?

Comment (AnthonySteele) [inline NuKeeper/GithubEngine.cs]: Of course. I'm hoping that all of that will eventually turn into IoC

**Comment by cohen990** [comment]
happy when you are @AnthonySteele :+1:

**Comment by AnthonySteele** [comment]
Yep, ready to merge now

**Pull request merged**